common/scala/src/main/scala/org/apache/openwhisk/core/entity/Identity.scala

```diff
@@ -35,12 +35,14 @@ case class UserLimits(invocationsPerMinute: Option[Int] = None,
                       concurrentInvocations: Option[Int] = None,
                       firesPerMinute: Option[Int] = None,
                       allowedKinds: Option[Set[String]] = None,
-                      storeActivations: Option[Boolean] = None)
+                      storeActivations: Option[Boolean] = None,
+                      warmedContainerKeepingCount: Option[Int] = None,
+                      warmedContainerKeepingTimeout: Option[String] = None)
```

Comment:
maybe have infects for non-scheduler codes?

Reply:
Yes，infects below codes
```
case class UserLimits(invocationsPerMinute: Option[Int] = None,
                      concurrentInvocations: Option[Int] = None,
                      firesPerMinute: Option[Int] = None,
                      allowedKinds: Option[Set[String]] = None,
                      storeActivations: Option[Boolean] = None,
                      warmedContainerKeepingCount: Option[Int] = None,        // this is the new field
                      warmedContainerKeepingTimeout: Option[String] = None)   // this is the new field
```

If don't want to infect above code, one option is
`make the keepingCount and keepingTime value` to a fixed value this time(don't read from db),
and submit the changes(e.g. add above 2 fileds) with another subsequent pr.

hm..but obviously, configure it in db is a better solution.

Reply:
@bdoyle0182 @style95 ,due to above 2 fields(warmedContainerKeepingCount, warmedContainerKeepingTimeout) are added into `case class UserLimits`, do you guys have any opinion?

Reply:
I'm not opposed to infecting non-scheduler code where absolutely necessary. I think it's unreasonable to suggest we can make such a large architectural change without touching existing code at all. So long as we're avoiding a breaking change I see no issue with this

Reply:
@ningyougang @style95 also sorry haven't had time to dedicate to the scheduler recently, but will review the remaining existing pr's this week

Reply:
@bdoyle0182 ,thanks for you review, just add 2 fields to `case class UserLimits`, it is not  `a large architectural change`

Reply:
Yea I just meant the entire new scheduler is `a large architectural change` so it's unreasonable to say we can't ever touch other components while implementing. We should avoid where possible, but it will be necessary in a couple situations like this one.